crates/re_renderer/src/renderer/lines.rs

```diff
@@ -438,155 +419,112 @@ impl LineDrawData {
-            let mut position_data_staging = Vec::with_capacity(wgpu::util::align_to(
-                num_segments,
-                POSITION_TEXTURE_SIZE,
-            ) as usize);
+            re_tracing::profile_scope!("write_pos_data_texture");
+
+            let texture_size = position_data_texture.texture.size();
+            let texel_count = (texture_size.width * texture_size.height) as usize;
+            let num_elements_padding = texel_count - num_line_vertices as usize;
+
+            let mut staging_buffer = ctx.cpu_write_gpu_read_belt.lock().allocate(
+                &ctx.device,
+                &ctx.gpu_resources.buffers,
+                texel_count,
+            )?;
+
             // sentinel at the beginning to facilitate caps.
-            position_data_staging.push(LineVertex {
+            staging_buffer.push(LineVertex {
                 position: glam::vec3(f32::MAX, f32::MAX, f32::MAX),
                 strip_index: u32::MAX,
-            });
-            position_data_staging.extend(vertices.iter());
+            })?;
+            staging_buffer.extend_from_slice(vertices)?;
             // placeholder at the end to facilitate caps.
-            position_data_staging.push(LineVertex {
+            staging_buffer.push(LineVertex {
                 position: glam::vec3(f32::MAX, f32::MAX, f32::MAX),
                 strip_index: u32::MAX,
-            });
-            position_data_staging.extend(std::iter::repeat(gpu_data::LineVertex::zeroed()).take(
-                (wgpu::util::align_to(num_segments, POSITION_TEXTURE_SIZE) - num_segments) as usize,
-            ));
-
-            // TODO(andreas): Use staging belt here.
-            ctx.queue.write_texture(
-                wgpu::ImageCopyTexture {
-                    texture: &position_data_texture.texture,
-                    mip_level: 0,
-                    origin: wgpu::Origin3d::ZERO,
-                    aspect: wgpu::TextureAspect::All,
-                },
-                bytemuck::cast_slice(&position_data_staging),
-                wgpu::ImageDataLayout {
-                    offset: 0,
-                    bytes_per_row: Some(
-                        POSITION_TEXTURE_SIZE * std::mem::size_of::<gpu_data::LineVertex>() as u32,
-                    ),
-                    rows_per_image: None,
-                },
-                wgpu::Extent3d {
-                    width: POSITION_TEXTURE_SIZE,
-                    height: (num_segments + POSITION_TEXTURE_SIZE - 1) / POSITION_TEXTURE_SIZE,
-                    depth_or_array_layers: 1,
-                },
-            );
-        }
+            })?;
+            staging_buffer.fill_n(gpu_data::LineVertex::zeroed(), num_elements_padding)?;
 
+            staging_buffer.copy_to_texture2d_entire_first_layer(
+                copy_encoder.lock().get(),
+                &position_data_texture,
+            )?;
+        }
         // Upload strip data.
         {
```

Comment:
nit: by moving the comment inside the block it makes it more likely that it will still be attached when moving code around
```suggestion
        {
            // Upload strip data.
```